Added validation check for integer value for series.df (#56688)
pandas-dev/pandas: doc/source/whatsnew/v2.3.0.rst

Bug fixes
~~~~~~~~~

> Fixed bug in :meth:`Series.diff` and :meth:`algorithms.diff` for no arg validation unlike :meth:`Dataframe.diff` for periods. (:issue:`56607`)

Reviewer: ```suggestion
Fixed bug in :meth:`Series.diff` and :meth:`algorithms.diff` allowing non-integer values for the ``periods`` argument. (:issue:`56607`)
```

Author: Shouldn't it be allowing only integer values @lithomas1 ?

Author: Done @lithomas1

Bug fixes
~~~~~~~~~

> Fixed bug in :meth:`Series.diff` and :meth:`algorithms.diff` allowing non-integer values for the ``periods`` argument. (:issue:`56607`)

Reviewer: ```suggestion
- Fixed bug in :meth:`Series.diff` and :meth:`algorithms.diff` allowing non-integer values for the ``periods`` argument. (:issue:`56607`)
```

You need to keep the ``-`` too I think to keep formatting consistent with the rest of the whatsnew entries.

> - Fixed bug in :meth:`Series.diff` and :meth:`algorithms.diff` allowing non-integer values for the ``periods`` argument. (:issue:`56607`)

Reviewer: ```suggestion
- Fixed bug in :meth:`Series.diff` allowing non-integer values for the ``periods`` argument. (:issue:`56607`)
```

One last comment. ``pandas.core.algorithms`` isn't part of the public API, so there's no need to mention it in the whatsnew.
